addons/website/models/website_page.py

```diff
@@ -113,15 +113,13 @@ def _get_most_specific_pages(self):
         return self.browse(ids)
 
     def copy_data(self, default=None):
-        vals_list = super().copy_data(default=default)
-        if not default:
-            return vals_list
-        for page, vals in zip(self, vals_list):
+        default = dict(default or {})
+        for page in self:
             if not default.get('view_id'):
                 new_view = page.view_id.copy({'website_id': default.get('website_id')})
-                vals['view_id'] = new_view.id
-            vals['url'] = default.get('url', self.env['website'].get_unique_path(page.url))
-        return vals_list
+                default['view_id'] = new_view.id
+            default['url'] = default.get('url', self.env['website'].get_unique_path(page.url))
+        return super().copy_data(default=default)
```

Comment:
As discussed, this is a solution but not the correct one IMO, because this will only work as long as only one record is copied, as the default dictionary is common to all records, which means all records will then share the same view_id

>>> Page = self.env['website.page']
>>> contact_pages = Page.search([('name', 'ilike', 'contact')])
>>> dup = contact_pages.copy({ 'website_id': 1 })
>>> dup[0].view_id
ir.ui.view(1199,)

>>> dup[1].view_id
ir.ui.view(1199,)

>>> contact_pages[0].view_id
ir.ui.view(501,)

>>> contact_pages[1].view_id
ir.ui.view(500,)